### PR TITLE
Added the channel argument

### PR DESCRIPTION
**Pull request by Taliik**
## Summary of the changes / Why this is an improvement
The channel argument is needed to be able to specify if the version should be a stable/testing/nightly version.

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -5,6 +5,9 @@ Changes for croud
 Unreleased
 ==========
 
+- Add the ``channel`` argument to the cluster deploy command. This is needed
```
Comment (chaudum) [inline CHANGES.rst]: Please write the changelog entry for the end-user:

```suggestion
- Added the ``channel`` argument to the ``clusters deploy`` command. This argument can be used to specify the release channel of the specified CrateDB version. Can be either ``stable`` (default), ``testing``, or ``nightly``. Testing and nightly deployments can only be performed by superusers. 
```

File: croud/cmd.py [hunk 2]
```diff
@@ -289,3 +289,16 @@ def num_instances_arg(req_args: _ArgumentGroup, opt_args: _ArgumentGroup) -> Non
...
+def channel_arg(
+    req_args: _ArgumentGroup, opt_args: _ArgumentGroup, required: bool = True
+) -> None:
+    group = req_args if required else opt_args
+    group.add_argument(
+        "--channel",
+        type=str,
+        help="The channel of the CrateDB version.",
+        default="stable",
+        required=False,
```
Comment (chaudum) [inline croud/cmd.py]:
```suggestion
        required=required,
```

File: croud/cmd.py [hunk 3]
```diff
@@ -289,3 +289,16 @@ def num_instances_arg(req_args: _ArgumentGroup, opt_args: _ArgumentGroup) -> Non
...
+    )
```
Comment (chaudum) [inline croud/cmd.py]: Please add possible choices:
```suggestion
        choices={"stable", "testing", "nightly"},
    )
```

File: tests/unit_tests/test_commands.py [hunk 4]
```diff
@@ -229,6 +230,50 @@ def test_deploy_cluster_no_unit(self, mock_send, mock_run, mock_load_config):
...
+            "--tier",
+            "xs",
+            "--unit",
+            "1",
+            "--project-id",
+            self.project_id,
+            "--cluster-name",
+            "crate_cluster",
+            "--version",
+            "nightly",
```
Comment (chaudum) [inline tests/unit_tests/test_commands.py]:
```suggestion
            "nightly-4.1.0-20190712",
```

File: croud/cmd.py [hunk 5]
```diff
@@ -289,3 +289,17 @@ def num_instances_arg(req_args: _ArgumentGroup, opt_args: _ArgumentGroup) -> Non
...
+
+
+def channel_arg(
+    req_args: _ArgumentGroup, opt_args: _ArgumentGroup, required: bool = True
+) -> None:
+    group = req_args if required else opt_args
+    group.add_argument(
+        "--channel",
+        type=str,
+        help="The channel of the CrateDB version.",
```
Comment (MarkusH) [inline croud/cmd.py]: If this is for superusers only, maybe add it to the help string?
```suggestion
        help="The channel of the CrateDB version (superusers only).",
```

**Pull request merged**